Cargo.toml

```diff
@@ -11,49 +15,69 @@ repository = "https://github.com/casper-ecosystem/casper-client-rs"
 license = "Apache-2.0"
 
 [lib]
+crate-type = ["cdylib", "rlib"]
 name = "casper_client"
 path = "lib/lib.rs"
 
 [[bin]]
 name = "casper-client"
 path = "src/main.rs"
 doc = false
+required-features = [
+    "async-trait",
+    "clap",
+    "clap_complete",
+    "tokio",
+    "std-fs-io",
+]
+
+[features]
+default = ["async-trait", "clap", "clap_complete", "tokio", "std-fs-io"]
+std-fs-io = ["casper-types/std-fs-io"]
 
 [dependencies]
-async-trait = "0.1.59"
+async-trait = { version = "0.1.74", optional = true }
 base16 = "0.2.1"
 casper-types = { version = "5.0.0", features = ["std"] }
-clap = { version = "4", features = ["cargo", "deprecated", "wrap_help"] }
-clap_complete = "4"
+clap = { version = "4.4.10", optional = true, features = [
+    "cargo",
+    "deprecated",
+] }
+clap_complete = { version = "4.4.4", optional = true }
 hex-buffer-serde = "0.4.0"
-humantime = "2"
-itertools = "0.11.0"
+humantime = "2.1.0"
+itertools = "0.12.0"
 jsonrpc-lite = "0.6.0"
 num-traits = "0.2.15"
-once_cell = "1"
+once_cell = "1.18.0"
 rand = "0.8.5"
-reqwest = { version = "0.11.13", features = ["json"] }
-schemars = "0.8.13"
-serde = { version = "1", default-features = false, features = ["derive"] }
+reqwest = { version = "0.12.4", features = ["json"] }
+schemars = "0.8.18"
+serde = { version = "1.0.193", default-features = false, features = ["derive"] }
 serde-map-to-array = "1.1.1"
-serde_json = { version = "1", features = ["preserve_order"] }
-thiserror = "1.0.34"
-tokio = { version = "1.23.0", features = ["macros", "net", "rt-multi-thread", "sync", "time", ]}
-uint = "0.9.4"
+serde_json = { version = "1.0.108", features = ["preserve_order"] }
+thiserror = "1.0.50"
+tokio = { version = "1.34.0", optional = true, features = [
+    "macros",
+    "rt",
+    "sync",
+    "time",
+] }
+uint = "0.9.5"
 
 [dev-dependencies]
-tempfile = "3.7.1"
+tempfile = "3.8.1"
 
 [build-dependencies]
 vergen = { version = "7", default-features = false, features = ["git"] }
 
 [patch.crates-io]
-casper-types = { git = "https://github.com/casper-network/casper-node", branch = "feat-2.0" }
+casper-types = { git = "https://github.com/casper-network/casper-node", branch = "rustSDK-feat-2.0" }
```

Comment:
This needs to be removed before merge, after casper-typs [PR](https://github.com/casper-network/casper-node/pull/4714)